.github/workflows/test-e2e-desktop.yaml

```diff
@@ -0,0 +1,109 @@
+name: Test E2E Desktop Client
+
+on:
+  pull_request:
+    types: [labeled]
+  workflow_dispatch:
+
+jobs:
+  test:
+    if: ${{ github.repository == 'hashicorp/boundary-ui' && github.event.label.name == 'desktop' }}
+    name: Run Desktop E2E Tests
+    runs-on: ubuntu-22.04
+    env:
+      ENOS_VAR_boundary_edition: "ent"
+      ENOS_VAR_e2e_debug_no_run: true
+      ENOS_VAR_boundary_license_path: "./license.hclic"
+      SETUP_CLI: true
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Node
+        uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
+        with:
+          node-version: 20
+          cache: "yarn"
+          cache-dependency-path: |
+            yarn.lock
+            ui/desktop/electron-app/yarn.lock
+
+      - name: Set up Terraform
+        uses: hashicorp/setup-terraform@a1502cd9e758c50496cc9ac5308c4843bcd56d36
+
+      - name: Set up Enos
+        uses: hashicorp/action-setup-enos@b9fa53484a1e8fdcc7b02a118bcf01d65b9414c9
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Vault CLI
+        run: |
+          wget https://releases.hashicorp.com/vault/1.12.2/vault_1.12.2_linux_amd64.zip -O /tmp/vault.zip
+          unzip /tmp/vault.zip -d /usr/local/bin
```

Comment:
We'll need to see if we want to figure out how to keep the version of vault in sync with the version of vault used in the docker container versus just hard-coding the version here